src/useCustomSelector.lua

```diff
@@ -4,14 +4,16 @@ end
 
 local function useCustomSelector(
 	hooks,
-	selector: (state: table) -> any,
-	equalityFn: ((newState: table, oldState: table) -> boolean)?,
+	selector: (state: any) -> any,
+	equalityFn: ((newState: any, oldState: any) -> boolean)?,
 	context
 )
+	-- This value wrapper is required so the variable context of the selector function can be updated on each run --
+	local selectorFunc = hooks.useValue()
+	selectorFunc.value = selector
+
 	local store = hooks.useContext(context)
-	local mappedState, setMappedState = hooks.useState(function()
-		return selector(store:getState())
-	end)
+	local mappedState, setMappedState = hooks.useState(selector(store:getState()))
```

Comment:
When you pass the selector function directly to `useState` like this it will call `selector` every re-render. You should probably wrap this in another function to avoid that. [I made the same mistake](https://discord.com/channels/385151591524597761/440326611863339009/936434638619021344) with the initial release of rodux-hooks.

Reply:
Interesting, noted! I've run into this before and had to do a side-effect-set workaround.